From 0b36c28fe8e3086d0c0d7e991a09f713e8c1aa5c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 13 May 2021 22:05:00 +0200
Subject: [PATCH] Specify required versions of GI components

This fixes transmageddon to run when GTK4 is installed on the system.
---
 src/transmageddon.py | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/transmageddon.py b/src/transmageddon.py
index 0721249..89aa36c 100644
--- a/src/transmageddon.py
+++ b/src/transmageddon.py
@@ -31,8 +31,16 @@ if sys.version_info[0] < 3:
 
 import which
 import time
+import gi
+gi.require_version('Notify', '0.7')
+gi.require_version('Gdk', '3.0')
+gi.require_version('Gtk', '3.0')
+gi.require_version('Gst', '1.0')
+gi.require_version('GstPbutils', '1.0')
+gi.require_version('GstTag', '1.0')
+gi.require_version('GUdev', '1.0')
 from gi.repository import Notify
-from gi.repository import GdkX11, Gdk, Gio, Gtk, GLib, Gst, GstPbutils, GstTag
+from gi.repository import Gdk, GdkX11, Gio, Gtk, GLib, Gst, GstPbutils, GstTag
 from gi.repository import GUdev
 from gi.repository import GObject, GdkPixbuf
 # GObject.threads_init()
-- 
2.31.1

